packages/react-core/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -280,13 +280,12 @@ export const CalendarMonth = ({
                 const isSelected = isValidDate(selectedDate) && isSameDate(date, selectedDate);
                 const isFocused = isSameDate(date, focusedDate);
                 const isAdjacentMonth = date.getMonth() !== focusedDate.getMonth();
+                const isRangeStart = isValidDate(rangeStart) && isSameDate(date, rangeStart);
                 let isInRange = false;
-                let isRangeStart = false;
                 let isRangeEnd = false;
                 if (isValidDate(rangeStart) && isHoveredDateValid) {
                   if (hoveredDate > rangeStart || isSameDate(hoveredDate, rangeStart)) {
```

Comment:
Sorry I noticed a slight issue with my original suggestion, 
![Mar-25-2022 08-59-10](https://user-images.githubusercontent.com/26305082/160129860-e6106b37-d160-4833-b93c-61c0adc766f6.gif)
That is the behavior I was suggesting.

but I think the code that needs to be updated is more like
on CalendarMonth.tsx line 116
```if (!isValidDate(focusedDate)) {
    if (isValidDate(rangeStart)) {
      focusedDate = rangeStart;
    } else {
      focusedDate = today;
    }
  }
```
That will focus on the start date if it’s provided and no date prop is set
and then on line 290
```
                if (isValidDate(rangeStart) && isHoveredDateValid) {
                  if (hoveredDate > rangeStart || isSameDate(hoveredDate, rangeStart)) {
                    if (isValidDate(selectedDate)) {
                      isInRange = date > rangeStart && date < selectedDate;
                      isRangeEnd = isSameDate(date, selectedDate);
                    } else {
                      isInRange = date > rangeStart && date < hoveredDate;
                      isRangeEnd = isSameDate(date, hoveredDate);
                    }

                  }
                  // Don't handle focused dates before start dates for now.
                  // Core would likely need new styles
                }
```